src/Control/PathfinderRequestHandler.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace CodeCraft\Pathfinder\Control;
+
+use CodeCraft\Pathfinder\Extension\PathfinderControllerExtension;
+use CodeCraft\Pathfinder\Model\Pathfinder;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\RequestHandler;
+
+/**
+ * A handler for the contiguous requests of a user proceeding through a Pathfinder
+ */
+class PathfinderRequestHandler extends RequestHandler
+{
+    /**
+     * @var array
+     */
+    private static $allowed_actions = [
```

Comment:
I don't have a clear picture on how `RequestHandlers` work - would you be able to expound on why the `reset` action couldn't just live in the `PathfinderPageController` class as well as all the need for the class itself and its associated methods? For module/customisation/future proofing purposes?

Reply:
`Controller`s are `RequestHandler`s, so it could be either or, and therefor this is more of a code organisation thing. 

Using our own case-specific `RequestHandler` means we can organise this logic at a nested level, rather than adding the boilerplate at the controller-level. Organising it this way means that it's less opinionated, and its concerns are self contained, which has some benefits e.g we can extend a block controller without any additional code, other than the extension. By contrast, if you checkout how dnadesign/silverstripe-elemental-userforms has to scaffold the `UserDefinedFormController`, rather than just a clean deligation.